app/services/payment_provider_customers/adyen_service.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module PaymentProviderCustomers
+  class AdyenService < BaseService
+    def initialize(adyen_customer = nil)
+      @adyen_customer = adyen_customer
+
+      super(nil)
+    end
+
+    def create
+      result.adyen_customer = adyen_customer
+      return result if adyen_customer.provider_customer_id?
+
+      adyen_result = generate_checkout_url
+
+      result.checkout_url = adyen_result.response["url"]
+      result
+    end
+
+    def generate_checkout_url
+      res = client.checkout.payment_links_api.payment_links(payment_link_params)
+
+      SendWebhookJob.perform_later(
+        'customer.checkout_url_generated',
+        customer,
+        checkout_url: res.response["url"]
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://rubystyle.guide#consistent-string-literals)